test/integration/component/test_persistent_networks.py

```diff
@@ -1459,6 +1459,135 @@ def test_delete_account(self):
 
         return
 
+    def test_volume_delete_event_errorState(self):
+        """
+        @summary: Test volume delete event generation in error state condition
+        @Steps:
+
+        Step1: Create  a network using network created in Step1
+        Step2: Verifying that  network creation is successful
+        Step3: Login to Virtual router and add iptable  rule to block insertion of vm rules
+        Step6: deploy a vm using network created in step2
+        Step7: check the Vm status for failure
+        Step8: destroy and expunge the vm
+        Step9: list the generated events for volume delete event.
+        """
+
+        # Listing all the networks available
+
+        account = Account.create(
+            self.api_client,
+            self.services["account"],
+            domainid=self.domain.id)
+
+        network = Network.create(
+            self.apiclient,
+            self.services["isolated_network"],
+            networkofferingid=self.isolated_persistent_network_offering.id,
+            accountid=self.account.name,
+            domainid=self.domain.id,
+            zoneid=self.zone.id)
+
+        response = verifyNetworkState(
+            self.apiclient,
+            network.id,
+            "implemented")
+        exceptionOccured = response[0]
+        isNetworkInDesiredState = response[1]
+        exceptionMessage = response[2]
+
+        if (exceptionOccured or (not isNetworkInDesiredState)):
+            self.fail(exceptionMessage)
+        self.assertIsNotNone(
+            network.vlan,
+            "vlan must not be null for persistent network")
+        try:
+            if self.zone.networktype == "Basic":
+                list_router_response = list_routers(
+                    self.apiclient,
+                    listall="true"
+                )
+            else:
+                list_router_response = list_routers(
+                    self.apiclient,
+                    account=self.account.name,
+                    domainid=self.account.domainid
+                )
+            self.assertEqual(
+                isinstance(list_router_response, list),
+                True,
+                "Check list response returns a valid list"
+            )
+            router = list_router_response[0]
+
+            self.debug("Router ID: %s, state: %s" % (router.id, router.state))
+
+            self.assertEqual(
+                router.state,
+                'Running',
+                "Check list router response for router state"
+            )
+            self.hypervisor = self.testClient.getHypervisorInfo()
+            if self.hypervisor.lower() in ('vmware', 'hyperv'):
+                result = get_process_status(
+                    self.apiclient.connection.mgtSvr,
+                    22,
+                    self.apiclient.connection.user,
+                    self.apiclient.connection.passwd,
+                    router.linklocalip,
+                    "iptables -I INPUT 1 -j DROP",
```

Comment:
Delete this rule once the test is done so that it won't have impact on the other tests.